pandas/io/common.py

```diff
@@ -520,6 +521,9 @@ def infer_compression(
             # Cannot infer compression of a buffer, assume no compression
             return None
 
+        if ".tar" in filepath_or_buffer:
```

Comment:
There are only four options for tar (.tar, tar.gz, tar.bz, tar.xz), or? Might be simpler to add them to `_compression_to_extension`.

Reply:
Makes sense! Implemented in https://github.com/pandas-dev/pandas/pull/44787/commits/6a8edef38437769491c645e284ec2696e13b8182 and https://github.com/pandas-dev/pandas/pull/44787/commits/d4e40c97b67ab0ad98e5435b0ef0710310fb032e.